src/services/platform.service.js

```diff
@@ -73,76 +73,37 @@ export const getBaseProjectEnvironment = (
   };
 };
 
-// HACK: With electron-builder, we're having some issues on mac finding Node.
-// This is because for some reason, the PATH is not updated properly :(
-// 'fix-path' is supposed to do this for us, but it doesn't work, for unknown
-// reasons.
+// HACK: With electron-builder, we're having some issues on Mac finding Node.
+// GUI apps on macOS (aka not launched from terminal) don't inherit environment variables
+// defined in dotfiles, so need to fix the path.
 export const initializePath = () => {
   return new Promise<void>(resolve => {
     if (!isMac) {
       return resolve();
     }
 
-    // Check if we need to fix the Path (Mac only)
+    // Check if Node is found
     childProcess.exec(
       'which node',
       { env: window.process.env },
       (_, nodePath) => {
         if (nodePath) {
-          // Node found
+          // If it is, no need to fix path
           return resolve();
         }
-
-        // Get login username
-        const username = os.userInfo().username;
-
-        childProcess.exec(
-          `dscl . -read /Users/${username} UserShell`,
-          (error, stdout) => {
-            if (error) {
-              resolve(error);
-            }
-
-            // Check which shell is in use e.g. /bin/bash or /bin/zsh
-            // Get the name of the shell after the last instant of '/'
-            const shell = stdout
-              .substring(stdout.lastIndexOf('/') + 1, stdout.length)
-              .trim();
-
-            // Format the config file using correct shell
-            let sourceFile =
-              shell === 'fish' ? '.config/fish/config.fish' : `.${shell}rc`;
-
-            // Check the file actually exists first
-            const fileExists = fs.existsSync(`${os.homedir()}/${sourceFile}`);
-
-            // If file doesn't exist then source .bash_profile instead
-            if (!fileExists) {
-              sourceFile = '~/.bash_profile';
-            }
-
-            // For users with a standard Node installation, node will be in
-            // /usr/local/bin
-            // For users using NVM, the path to Node will be added to the shell config file.
-            // Add both to the PATH.
-            try {
-              childProcess.exec(
-                `source ~/${sourceFile} ; echo $PATH`,
-                (err, updatedPath) => {
-                  if (updatedPath) {
-                    window.process.env.PATH = `/usr/local/bin:${updatedPath}`;
-                  }
-
-                  resolve();
-                }
-              );
-            } catch (e) {
-              resolve(e);
-            }
-          }
-        );
       }
     );
+
+    // This is basically `fix-path` but we are adding it to window instead
+    window.process.env.PATH =
```

Comment:
I tried to use fixPath here but it was not working at all. The only difference we add it to the window object here, whereas fixPath adds it to `process.env.PATH`.